openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -294,3 +292,85 @@ def post(self, request, username):
             return Response(text_type(exc), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class DeactivateLogoutView(APIView):
+    """
+    POST /api/user/v1/accounts/deactivate_logout/
+    {
+        "user": "example_username",
+    }
+
+    **POST Parameters**
+
+      A POST request must include the following parameter.
+
+      * user: Required. The username of the user being deactivated.
+
+    **POST Response Values**
+
+     If the request does not specify a username or submits a username
+     for a non-existent user, the request returns an HTTP 404 "Not Found"
+     response.
+
+     If a user who is not a superuser tries to deactivate a user,
+     the request returns an HTTP 403 "Forbidden" response.
+
+     If the specified user is successfully deactivated, the request
+     returns an HTTP 204 "No Content" response.
+
+     If an unanticipated error occurs, the request returns an
+     HTTP 500 "Internal Server Error" response.
+
+    Allows an administrative user to take the following actions
+    on behalf of an LMS user:
+    -  Change the user's password permanently to Django's unusable password
+    -  Log the user out
+    """
+    authentication_classes = (JwtAuthentication, )
+    permission_classes = (permissions.IsAuthenticated, CanRetireUser)
+
+    def post(self, request):
+        """
+        POST /api/user/v1/accounts/deactivate_logout
+
+        Marks the user as having no password set for deactivation purposes,
+        and logs the user out.
+        """
+        username = request.data.get('user', None)
+        if not username:
+            return Response(
+                status=status.HTTP_404_NOT_FOUND,
+                data={
+                    'message': u'The user was not specified.'
+                }
+            )
+
+        user_model = get_user_model()
+        try:
+            # make sure the specified user exists
+            user = user_model.objects.get(username=username)
+
+            with transaction.atomic():
+                # 1. Unlink LMS social auth accounts
+                UserSocialAuth.objects.filter(user_id=user.id).delete()
+                # 2. Change LMS password & email
+                user.email = get_retired_email_by_email(user.email)
+                user.save()
```

Comment:
You could eliminate this call, since `_set_unusable_password(user)` will call `user.save()`.  The other option would be to get rid of the `set_unusable_password()` call and just call `user.set_unusable_password()` directly, then call `user.save()`.  I'm not sure if it really matters that `save()` is called twice here, though.